feat(design): Added jump-to-state combobox

kevee opened the pull request:
Related to #325 

This adds a Combobox element to the right/bottom of the data page that jumps people to a selected state.

The existing state nav element is retained and hidden except with JS turned off.

beep commented:
A few things I noticed!

- I’m not quite clear on how the sorting works? The first result was often unrelated to the letter I typed in. (Here’s [a screenshot for `m`](https://user-images.githubusercontent.com/124493/78376254-0bb54b00-759c-11ea-92e0-16c59a8b69c4.png) and [a screenshot for `c`](https://user-images.githubusercontent.com/124493/78376242-0821c400-759c-11ea-8ecd-14304bdd49d6.png).) (If the two-letter abbreviations are the culprit, I think we could probably drop them.)
- Can “Jump to state” be a `label` associated with the rendered field?
- Three usability questions!
  - I wonder if the element could better describe the action required? Like, could/should the label read “Type a state’s name to jump to it”?
  - Is there any way to get the full list to appear when the text field first gains focus?
  - If not, it’d be really great if pressing the down arrow revealed the full set of options. (Sort of how [the native `datalist`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist) element works.)
- Smol feature request: could we make the module containing “Totals by state” and the jump menu sticky?

kevee commented:
OK, I:

Removed the state code, that was causing errors
Made the box always appear and boldify the found text
Changed to a label element for the input
Changed the label text

kevee commented:
...and using a sticky head covers the title of the state when you jump to its anchor...  🤔

kevee commented:
Yes it would!

schwartzadev commented on src/pages/data/index.js:
```diff
@@ -34,19 +44,110 @@ const StateList = ({ states, stateData }) => {
+      >
+        <ComboboxInput
+          id="jump-to-state"
+          placeholder="State or territory"
```
:+1: for not using "state" :smile:

kevee replied:
Having lived in a territory, I try not to be "state normative."

kevee commented:
OK, updates:

- Switched to searching as an autocomplete (start of state name) rather than any match (i.e. "al" matches "C**al**ifornia)
- Disabled browser autocomplete
- `scroll-margin` was not working when updating the window location with Javascript. Used an older trick to not cover state headers when navigating.

Note that we don't have a sticky header in mobile because it was taking up valuable real-estate. I defer to other's judgement on that.

schwartzadev commented:
@kevee that was the issue with JS -- thanks 👍 

Yay! The autocomplete algorithm looks good -- this is a great feature to add!

My only suggestion at this point is pretty minor: when there is only one result remaining, I expect that pressing `enter` would jump to it. (Similar to navigating HTML's `select` with a keyboard.) However, that's not the current behavior; right now, pressing enter does nothing. (i.e. try entering `ca`. The only result is California, yet pressing enter does not jump to California.)

This is _super_ marginal -- I'm on board if you'd like to merge and move this to an issue instead of waiting to merge.

kevee commented:
@schwartzadev - Thanks for the feedback! I just added your suggestion about Enter key when there is one result also jumping you to that result.

schwartzadev commented:
Yes! This looks phenomenal 😄 The only thing that I found as an issue was the nav bar padding seems to break off on the right side between 1080px and 640px:

![image](https://user-images.githubusercontent.com/18607205/78746653-ac9d6080-7924-11ea-9088-d95fe0de6e81.png)

I'll see if I can fix it 👍

schwartzadev commented:
Hmm it looks like it's because of the container width but the tables scroll at that width...
![image](https://user-images.githubusercontent.com/18607205/78746922-554bc000-7925-11ea-87e4-fcf1f586ff68.png)

schwartzadev commented:
b79cce1145bfde2affb4500ab8748f0865e4bfa4 helps this but extends the bottom border to the right edge. I'll look into a fix.
![image](https://user-images.githubusercontent.com/18607205/78748573-99d95a80-7929-11ea-9c7c-de8fef22e361.png)

schwartzadev commented:
...right when I thought I fixed this...the table borders are peeking out:
![image](https://user-images.githubusercontent.com/18607205/78750127-d78bb280-792c-11ea-9d55-feb6fdae8ffd.png)

kevee commented:
Thanks for all the helpful tweaks, @schwartzadev !
